WordPress/src/main/java/org/wordpress/android/ui/main/MainBottomSheetFragment.kt

```diff
@@ -57,10 +72,71 @@ class MainBottomSheetFragment : BottomSheetDialogFragment() {
                 behavior.state = BottomSheetBehavior.STATE_EXPANDED
             }
         }
+
+        if (quickStartEvent != null) {
+            showQuickStartFocusPoint()
+        }
     }
 
     override fun onAttach(context: Context) {
         super.onAttach(context)
         (requireActivity().applicationContext as WordPress).component().inject(this)
     }
+
+    @Subscribe(sticky = true, threadMode = ThreadMode.MAIN)
+    @SuppressWarnings("unused")
+    fun onEvent(event: QuickStartEvent) {
+        if (!isAdded || view == null) {
+            return
+        }
+
+        EventBus.getDefault().removeStickyEvent(event)
+        quickStartEvent = event
+
+        if (quickStartEvent?.task == QuickStartTask.PUBLISH_POST) {
+            showQuickStartFocusPoint()
+        }
+    }
+
+    private fun showQuickStartFocusPoint() {
+        // we are waiting for RecyclerView to populate itself with views and then grab the one we need one when it's ready
+        content_recycler_view?.viewTreeObserver?.addOnGlobalLayoutListener(object : OnGlobalLayoutListener {
```

Comment:
I think the correct solution with the MVVM would be to include the information whether the quick start is showing into the `MainActionListItem` object. This way you'd only have to call something like `viewModel. showQuickStartFocusPoint` here, this would update the live data with the main actions and you can observe the results and show the point there. The view holder in the recycler view should the quick start point if the `MainActionListItem` object has the flag set. Does it make sense? If you do it this way, you don't need to use the `addOnGlobalLayoutListener ` or find the view holder, all will be automatic.

Reply:
I absolutely agree that the correct implementation should utilize MVVM. I learned so much since the time Quick Start was implemented, so I would rewrite it all in a heartbeat (current implementation is not cool) :) 

While putting the logic into view holder, MainActionListItem, and ViewModel is the right way to go, I'm not inclined to do it with the current implementation of Quick Start, since it was built to be as non-intrusive and as decoupled as possible. I would like to keep the logic consistent between different QS tours, so in case we would like to scrap, replace, or modify it, it would be pretty straightforward (so we won't have some parts baked into ViewModel and other manipulating views from above).

Reply:
First of all let me say I trust your judgement so feel free to merge this PR 👍 
However, consider a few things:
- I think it's OK to introduce a new, better way of implementing QuickStart that's not consistent and use this method in all the places where we use view models
- This is a new part of code, with view models and Kotlin. Working around the MVVM flow tells the future devs it's ok to do this when it's not. 
- The GlobalLayoutListener approach is a hack around the RecyclerView
- QuickStart in the viewmodel could be completely extracted. What you'd need is a boolean flag on the model itself. You could have an injected class that tells you `shouldShowQuickStartOnBottomSheet` to populate the flag. You already handle the clicks. 
- The future devs will be facing the same dilemma - clean and inconsistent or hacky and consistent

A I said, feel free to merge this PR. Maybe create a task to implement QS in a ViewModel?

Reply:
@planarvoid You are totally right. Especially given all the talk about tech debt :) Not the best idea to go around proper implementation, even if it's justified in one way or another. I'll integrate it into an existing flow, and submit a commit shortly.